File: atosim/org.palladiosimulator.blockchainsystems.threesim/src/main/java/org/palladiosimulator/blockchainsystems/threesim/metrics/FinneyAttackSuccess.java
package org.palladiosimulator.blockchainsystems.threesim.metrics;

import org.palladiosimulator.blockchainsystems.threesim.metrics.abstractions.OutputMetric;

public class FinneyAttackSuccess implements OutputMetric<Integer> {

    public static final String NAME = "Finney Attack Success";

    private final boolean _succeeded;

    public FinneyAttackSuccess(boolean succeeded) {
        _succeeded = succeeded;
    }

    @Override public Integer getValue() { return _succeeded ? 1 : 0; }
    @Override public String getName() { return NAME; }
    @Override public String getUnit() { return null; }
}

File: atosim/org.palladiosimulator.blockchainsystems.threesim/src/main/kotlin/org/palladiosimulator/blockchainsystems/threesim/metrics/FinneyAttackSuccess.kt
package org.palladiosimulator.blockchainsystems.threesim.metrics

import kotlinx.serialization.Serializable
import org.palladiosimulator.blockchainsystems.threesim.metrics.abstractions.OutputMetric

@Serializable
class FinneyAttackSuccess(
    private val succeeded: Boolean
) : OutputMetric<Int> {

    companion object {
        const val NAME = "Finney Attack Success"
    }

    override val value: Int
        get() = if (succeeded) 1 else 0

    override val name: String = NAME
    override val unit: String? = null
}